src/stapi_fastapi/backends/product_backend.py

```diff
@@ -16,9 +17,11 @@ async def search_opportunities(
         product_router: ProductRouter,
         search: OpportunityRequest,
         request: Request,
-    ) -> ResultE[list[Opportunity]]:
+        next: str | None,
```

Comment:
`next` and `limit` shouldn't be params here -- they should be part of the POST body, to correspond with the changes you previously made to the Link definition. As defined, these are query parameters rather than part of the POST body.  They can still be extracted from the body automatically, see https://fastapi.tiangolo.com/tutorial/body-multiple-params/#singular-values-in-body for how this is supposed to be done in the method passed to add_api_route, in this case ProductRouter#search_opportunities.

I think this also means there needs to be a change to the tests so that the value of the next link is used to page rather than manually construction the next request, since doing that should have revealed that these were query params instead of part of the body.

Reply:
Though I would also consider that next and limit should be part of the OpportunityRequest (and other *Request bodies)